### ISSUE TITLE
Mesa GBM regression on Linux

### ISSUE THREAD

**Issue by strassek** [intel/IA-Hardware-Composer#351]
Steps to reproduce:
1. Configure Mesa and IA-Hardware-Composer with '--enable-gbm'.
2. Run ./testlayers -j jsonconfigs/kmscube1layer.json
3. Check that demo doesn't start and results in this error:
https://github.com/intel/IA-Hardware-Composer/blob/0908c118bf7441d3c732ea81afdca18e7acf9015/os/linux/gbmbufferhandler.cpp#L344

My bisect resulted in a series of commits which couldn't be tested individually due to build breakages...

There are only 'skip'ped commits left to test.
The first bad commit could be any of:
45e7bb0678598992c7086d902d02db2d4c883178
879ada661af0ace1c721cb3e9e87cdeb0196b2da
448917d6623ed07091e0f3a1efa5ced31fcd50d4
d6d0eb59f94c7b56e586132541f7a847d2d18c27
b281680fef45e35a2cce201f8052872170020f37
We cannot bisect more!

**Comment by strassek** [comment]
@harishkrupo interesting, its continues to reproduce reliably for me.
I'm testing on Intel(R) Core(TM) i5-6600K CPU @ 3.50GHz with Ubuntu 16.04

libdrm configuration:
```
./autogen.sh --disable-radeon --disable-nouveau --disable-amdgpu --enable-udev --enable-libkms --prefix=$WLD
```
Mesa configuration:
```
./autogen.sh --prefix=$WLD --with-platforms=surfaceless --disable-dri3 --enable-shared-glapi --enable-texture-float --disable-glx --disable-glx-tls --enable-gbm --with-gallium-drivers= --with-dri-drivers=i965 --with-vulkan-drivers=intel
```
IA-Hardware-Composer configuration:
```
./autogen.sh --prefix=$WLD --enable-gbm
```

**Comment by strassek** [comment]
I have to assume this FIXME is relevant:
https://github.com/intel/IA-Hardware-Composer/blob/c646f38270ea0c7eed71d10bacd37b06d9860551/os/linux/gbmbufferhandler.cpp#L160-L161

**Comment by strassek** [comment]
I'll take a closer look at whats going wrong with RBC for Mesa GBM

**Comment by strassek** [comment]
Fixed in ea400bcd39db2d57035e99dc661f08b183de55bc
